helix-view/src/document.rs

```diff
@@ -124,6 +125,7 @@ impl Document {
             changes,
             old_state,
             diagnostics: Vec::new(),
+            current_diagnostic: 0,
```

Comment:
I think you shouldn't store an explicit position, instead "next" should be next diagnostic after the current cursor position (`doc.selection(view.id).cursor()`).

Reply:
I do like your approach better, updated.